auth-api/src/auth_api/services/reset.py

```diff
@@ -0,0 +1,39 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service for reset test data."""
+
+from auth_api.models import User as UserModel
+from auth_api.models import db
+
+
+class ResetTestData:  # pylint:disable=too-few-public-methods
+    """Cleanup all the data from model by created_by column."""
+
+    def __init__(self):
+        """Return a reset test data service instance."""
+
+    @staticmethod
+    def reset(token):
+        """Cleanup all the data from all tables create by the provided user id."""
+        user = UserModel.find_by_jwt_token(token)
+        if user:
+            for model_class in db.Model._decl_class_registry.values():  # pylint:disable=protected-access
```

Comment:
Is there a better way to do this? Not ideal accessing protected members unless we have to.  I think it's possible to get this info from the metadata.  https://www.pythonsheets.com/notes/python-sqlalchemy.html#reflection-loading-table-from-existing-database.  If not, no worries, this solution is good.

Reply:
I tried to use metadata to make this feature work. However, it still need to access the protected member to get the model class. We can use the metadata object to do query or delete but it won't pick up the relationships that defined in model class. I added a TODO tag of this so that we can go back to find a solution in the future.

Reply:
Sounds great, @pwei1018